runtime-sdk/src/crypto/signature/ed25519.rs

```diff
@@ -118,33 +118,29 @@ impl From<PublicKey> for CorePublicKey {
 
 /// A memory-backed signer for Ed25519.
 pub struct MemorySigner {
-    sk: ed25519_dalek::ExpandedSecretKey,
+    sk: ed25519_dalek::SigningKey,
 }
 
 impl MemorySigner {
     pub fn sign_digest<D>(&self, digest: D) -> Result<Signature, Error>
     where
         D: ed25519_dalek::Digest<OutputSize = U64>,
     {
-        let pk = ed25519_dalek::PublicKey::from(&self.sk);
         self.sk
-            .sign_prehashed(digest, &pk, None)
+            .sign_prehashed(digest, None)
             .map_err(|_| Error::SigningError)
             .map(|sig| sig.to_bytes().to_vec().into())
     }
 }
 
 impl super::Signer for MemorySigner {
     fn new_from_seed(seed: &[u8]) -> Result<Self, Error> {
-        let sk = ed25519_dalek::SecretKey::from_bytes(seed).map_err(|_| Error::InvalidArgument)?;
-        let esk = ed25519_dalek::ExpandedSecretKey::from(&sk);
-        Ok(Self { sk: esk })
+        Self::from_bytes(seed)
     }
 
     fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
         Ok(Self {
-            sk: ed25519_dalek::ExpandedSecretKey::from_bytes(bytes)
-                .map_err(|_| Error::MalformedPrivateKey)?,
+            sk: bytes.try_into().map_err(|_| Error::MalformedPrivateKey)?,
```

Comment:
I believe this (and corresponding `to_bytes` due to changed `sk` type) is breaking in case someone stored the result (key) in storage and/or is expecting the key format to be the old one. Not sure why we previously returned the expanded key (which is the wrong thing to do), I guess these weird ed25519 key variants contributed to this not being cought during review. :facepalm:

Reply:
that's okay, I can make it backwards compatible and add a test for it